src/domain/registration/query.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  QueryClient,
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from 'react-query';
+
+import { Registration, RegistrationQueryVariables } from './types';
+import { fetchRegistration } from './utils';
+
+/* istanbul ignore next */
+export const fetchRegistrationQuery = (
+  queryClient: QueryClient,
+  args: RegistrationQueryVariables
+): Promise<Registration> => {
+  return queryClient.fetchQuery('registration', () => fetchRegistration(args));
```

Comment:
Does caching work here when id is not given as the first parameter? `queryClient.fetchQuery(['registration', args.id]...`

Reply:
https://react-query.tanstack.com/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key

Reply:
I think it worked only because there was no other event/registration queries. Added event/registration id to cache key to make sure that it will work also in the future